### ISSUE TITLE
Scaffolded items not appearing in project

### ISSUE THREAD

**Issue by bradwebber** [processedbeets/ASP.NET-MVC-Scaffolding#10]
I have been trying to created scaffolded items for a couple of days now, but I'm not having any luck. I have tried scaffolding a controller and also tried the custom scaffolder. after entering the commands in the package manager console it returns successfully but no items have been added to my solution. I don't get any output in the package manager console either. For example, when trying the custom scaffolder I would expect the following output:

Added scaffolder script 'CodeTemplates\Scaffolders\Example\Example.ps1'
Added scaffolder template 'CodeTemplates\Scaffolders\Example\ExampleTemplate.cs.t4'

However I don't get any output at all.

I am using VS2017 Community on Windows 10 if that helps.

**Comment by cigano** [comment]
@bradwebber There's a file in the project root called `scaffolders.config`. What is its contents?

**Comment by bradwebber** [comment]
```
<?xml version="1.0" encoding="utf-8"?>
<Config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <DefaultScaffolders>
    <Default DefaultName="Action" ScaffolderName="MvcScaffolding.Action" />
    <Default DefaultName="Controller" ScaffolderName="MvcScaffolding.Controller" />
    <Default DefaultName="CustomScaffolder" ScaffolderName="T4Scaffolding.CustomScaffolder" />
    <Default DefaultName="CustomTemplate" ScaffolderName="T4Scaffolding.CustomTemplate" />
    <Default DefaultName="DbContext" ScaffolderName="T4Scaffolding.EFDbContext" />
    <Default DefaultName="Repository" ScaffolderName="T4Scaffolding.EFRepository" />
    <Default DefaultName="UnitTest" ScaffolderName="MvcScaffolding.ActionUnitTest" />
    <Default DefaultName="View" ScaffolderName="MvcScaffolding.RazorView" />
    <Default DefaultName="Views" ScaffolderName="MvcScaffolding.Views" />
  </DefaultScaffolders>
</Config>
```

scaffolders.config is actually appearing in my solution root folder rather than my project root folder.

I have done a bit more digging and it seems that the issue might be related to `Add-ProjectItemViaTemplate`. It looks like the scripts are successfully calling that commandlet, but it doesn't actually do anything.

For example, I modified the T4Scaffolding.CustomScaffolder.ps1 script to display a bit more information in the console:

```
[T4Scaffolding.Scaffolder(Description = "Creates an entirely new scaffolder with a PS1 script and a T4 template")][CmdletBinding()]
param(        
    [parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true)][string]$CustomScaffolderName,
    [string]$Project,
	[string]$CodeLanguage,
	[string[]]$TemplateFolders,
	[switch]$Force = $false
)

$templateName = $CustomScaffolderName + "Template"

Write-warning $Project
Write-Warning $CodeLanguage
Write-Warning "$TemplateFolders"
Write-Warning $templateName

# PS1 script
$customScaffoldersPath = [T4Scaffolding.Core.ScaffoldingConstants]::CustomScaffoldersFolderPath
Write-Warning $customScaffoldersPath

$outputPath = Join-Path (Join-Path $customScaffoldersPath $CustomScaffolderName) $CustomScaffolderName
Write-Warning $outputPath

Add-ProjectItemViaTemplate $outputPath -Template DefaultPs1Script -Model @{
	Scaffolder = $CustomScaffolderName; 
	TemplateName = $templateName;
} -SuccessMessage "Added scaffolder script '{0}'" -TemplateFolders $TemplateFolders -Project $Project -CodeLanguage "ps1" -Force:$Force

Write-Warning "Called Add-ProjectItemViaTemplate"

# T4 template
$outputPath = Join-Path (Join-Path $customScaffoldersPath $CustomScaffolderName) $templateName
Add-ProjectItemViaTemplate $outputPath -Template DefaultT4Template -Model @{
} -SuccessMessage "Added scaffolder template '{0}'" -TemplateFolders $TemplateFolders -Project $Project -CodeLanguage $CodeLanguage -Force:$Force

Write-Warning "Called Add-ProjectItemViaTemplate"
```
When I run `Scaffold CustomScaffolder Test` I get the following output:
`Called Add-ProjectItemViaTemplate`
`Called Add-ProjectItemViaTemplate`
but I don't see `Added scaffolder script 'DefaultPs1Script'` or `Added scaffolder template 'DefaultT4Template'`.

**Comment by cigano** [comment]
Yes, it seems to be a problem with the invocation. There's an extension for VS that allows you debugging the code: https://marketplace.visualstudio.com/items?itemName=AdamRDriscoll.PowerShellToolsforVisualStudio2015

**Comment by zhelyazkovn** [comment]
Hey, guys, I am fighting the exact same problem here - ps1 scripts working perfect on VS2015 but not on VS2017.  Add-ProjectItemViaTemplate does not create any files and does not throw any error, no where. 

Did you manage to fix this?

**Comment by processedbeets** [comment]
Hi @zhelyazkovn I've just merged a couple fixes from @BryanTheCrow and @cigano (many thanks tho you both for your contributions) Please grab the latest to see if this is still an issue.

**Comment by zhelyazkovn** [comment]
Thanks for the quick response. Yes I got the latest and there is no success :(

**Comment by processedbeets** [comment]
@cigano Forked this code and created a VS2017 version. Please check the ReadMe for links to those packages. You'll have to remove the VS2015 packages first:

https://github.com/processedbeets/ASP.NET-MVC-Scaffolding

thanks

**Comment by zhelyazkovn** [comment]
Yes, I went through the Read.me, I removed the 2015 ones and keep only 2017 ones. I also checked the last three commits, in order to understand more about the fixes, but all looks good my end  -I have  the latest - T4.Scaffolding.Core.VS2017 v.1.0.1. I updated the VS to the latest possible, and did all Window Updates. It has been three days I fight with this. Any help is welcome. Thank you!

**Comment by processedbeets** [comment]
Aw sorry to hear that. Alas, I really am not supporting this anymore, I can’t afford the time. It's been so long since I've studied the code, it would take me too long to get back up to speed.

There's a useful tool here that should help debugging within the code:
https://marketplace.visualstudio.com/items?itemName=AdamRDriscoll.PowerShellToolsforVisualStudio2017-18561

Good luck. If you do spot a fix I'd be happy to merge back in or you're welcome to fork and take this project further.

All the best

**Comment by zhelyazkovn** [comment]
Thanks mate, I understand! If I get it resolved will update the thread.

All the best
